rust/main/submitter/src/payload_dispatcher/stages/finality_stage.rs

```diff
@@ -1 +1,533 @@
+// TODO: re-enable clippy warnings
+#![allow(dead_code)]
 
+use std::{
+    collections::{HashMap, VecDeque},
+    future::Future,
+    sync::Arc,
+    time::Duration,
+};
+
+use derive_new::new;
+use eyre::{eyre, Result};
+use futures_util::future::try_join_all;
+use tokio::{
+    sync::{mpsc, Mutex},
+    time::sleep,
+};
+use tracing::{error, info, info_span, warn, Instrument};
+
+use crate::{
+    chain_tx_adapter::DispatcherError,
+    payload::{DropReason, FullPayload, PayloadStatus},
+    payload_dispatcher::stages::utils::update_tx_status,
+    transaction::{DropReason as TxDropReason, Transaction, TransactionId, TransactionStatus},
+};
+
+use super::{
+    building_stage::BuildingStageQueue, utils::retry_until_success, PayloadDispatcherState,
+};
+
+pub type FinalityStagePool = Arc<Mutex<HashMap<TransactionId, Transaction>>>;
+
+#[derive(new)]
+struct FinalityStage {
+    pool: FinalityStagePool,
+    inclusion_stage_receiver: mpsc::Receiver<Transaction>,
```

Comment:
I was a little confused by the name `inclusion_stage_receiver`. I understand that it is a receiver which expects transactions from inclusion stage. I suppose finality stage can receive transactions either from inclusion stage or database. 

I wonder if it makes sense to call it just a `receiver` as a field while it will be `finality_stage_receive` when struct `FinalityStage` is created?